services/search.service.js

```diff
@@ -63,15 +64,84 @@ function executeQuery(model, queryArray, page, limit, sort, sort_by, shouldExpan
     }
 
     if (sort == "desc") {
-        query.sort("-" + sort_by);
+        query.sort("-" + sortBy);
     } else if (sort == "asc") {
-        query.sort(sort_by);
+        query.sort(sortBy);
+    }
+    return query.limit(limit).skip(limit * page)
+}
+
+
+/**
+ * @function executeQuery
+ * @param {string} model the model which is being searched
+ * @param {Array} queryArray array of clauses for the query
+ * @param {number} page the page number you want
+ * @param {number} limit the limit to the number of responses you want
+ * @param {"asc"|"desc"} sort which direction you want to sort by
+ * @param {string} sortBy the attribute you want to sort by
+ * @returns {Promise<[Array]>}
+ * @description Builds and executes a search query based on a subset of mongodb
+ */
+function executeQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand = false) {
+    var query = createQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand);
+    return query.exec('find');
+}
+
+
+/**
+ * @function executeStatusAction
+ * @param {string} model the model which is being searched
+ * @param {Array} queryArray array of clauses for the query
+ * @param {number} page the page number you want
+ * @param {number} limit the limit to the number of responses you want
+ * @param {"asc"|"desc"} sort which direction you want to sort by
+ * @param {string} sortBy the attribute you want to sort by
+ * @param {string} update the JSON string containing the keys and values to update to
+ * @returns {Promise<[Array]>}
+ * @description Builds and executes a status update based on a subset of mongodb
+ */
+function executeStatusAction(model, queryArray, page, limit, sort, sortBy, update, shouldExpand = false) {
+    var query = createQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand);
+    var update_obj = JSON.parse(update);
+    return query.updateMany({ $set: update_obj }).exec();
+}
+
+
+/**
+ * @function executeEmailAction
+ * @param {string} model the model which is being searched
+ * @param {Array} queryArray array of clauses for the query
+ * @param {number} page the page number you want
+ * @param {number} limit the limit to the number of responses you want
+ * @param {"asc"|"desc"} sort which direction you want to sort by
+ * @param {string} sortBy the attribute you want to sort by
+ * @param {"Accepted"|"Waitlisted"|"Reminder"} status the status type of the email to send to hackers
+ * @returns {Promise<[Array]>}
+ * @description Sends a status update email based on a subset of mongodb
+ */
+async function executeEmailAction(model, queryArray, page, limit, sort, sortBy, status, shouldExpand = false) {
+    var query = createQuery(model, queryArray, page, limit, sort, sortBy, shouldExpand);
```

Comment:
This logic should not necessarily be in the executeEmailAction method, or at least if it is, then the parameters should reflect some sort of restrictions. For example, the code below assumes that the model is of type Hacker. However, the method parameters take in a `model` object! 

I would recommend this function be used only for Hacker objects, since it send a status update email. Because of this, the function name should change to reflect this.